src/api/app/controllers/webui/repositories_controller.rb

```diff
@@ -148,14 +148,16 @@ def create_dod_repository
   # POST project/create_image_repository
   def create_image_repository
     authorize @project, :update?
-    repository = @project.repositories.create!(name: 'images')
-    Architecture.available.each do |architecture|
-      repository.repository_architectures.create!(architecture: architecture)
-    end
-    @project.prepend_kiwi_config
+    repository = @project.repositories.new(name: 'images')
```

Comment:
What about: 
```
repository = @project.repositories.find_or_create_by(name: 'images')
```

And don't move the other parts instead?

Reply:
Well, this would also be a solution but then the response-message is not correct because it would always respond with "Successfully added image repository" even if it exists. So I'd prefer keeping it like this.

Reply:
You could check for ``new_record?`` for instance.

Reply:
@ChrisBr 

> You could check for ``new_record?`` for instance. 

And what is the difference in using `find_or_create` + `new_record?` and `new` + `save`? You are doing the same thing and what you want to do is to create a new repository and check if it is possible to save it. I would live it as @mschnitzer has already written it. The alternative for me obfuscate the code. 😕